frontend/src/components/ChangeUsername/changeusername.jsx

```diff
@@ -7,7 +7,7 @@ const ChangeUsername = () => {
 
     const handleSubmit = async (event) =>{
         event.preventDefault();
-        const response = await axios.post('/changeUsername', {
+        const response = await axios.post('/changeusername', {
             newUsername
           });

```

Comment:
now that you have cookies and session data stored in the client's browser, make sure to send that cookie by adding the `{ withCredentials: true }` option to every request you make to the server.
```suggestion
        const response = await axios.post('/changeusername', {
            newUsername
          }, { 
          withCredentials: true
          });
```